baybe/symmetries.py

```diff
@@ -0,0 +1,439 @@
+            if all(p.is_discrete and not p.is_numerical for p in params):
+                from baybe.parameters.base import DiscreteParameter
+
+                ref_vals = set(cast(DiscreteParameter, params[0]).values)
+                if any(
+                    set(cast(DiscreteParameter, p).values) != ref_vals for p in params
+                ):
+                    raise ValueError(
+                        f"The parameter group '{group}' contains parameters which have "
+                        f"different values. All parameters in a group must have the "
+                        f"same specification."
+                    )
+
+
+@define(frozen=True)
+class MirrorSymmetry(Symmetry):
+    """Class for representing mirror symmetries.
+
+    A mirror symmetry expresses that certain parameters can be inflected at a mirror
+    point without affecting the outcome of the model. For instance, this is the case if
+    $f(x,y) = f(-x,y)$ (mirror point is 0).
+    """
+
+    _parameter_name: str = field(validator=instance_of(str), alias="parameter_name")
+    """The name of the single parameter affected by the symmetry."""
+
+    # object variables
+    mirror_point: float = field(
+        default=0.0, converter=float, validator=validate_is_finite, kw_only=True
+    )
+    """The mirror point."""
+
+    @override
+    @property
+    def parameter_names(self) -> tuple[str]:
+        return (self._parameter_name,)
+
+    @override
+    def augment_measurements(
+        self, df: pd.DataFrame, _: Iterable[Parameter]
+    ) -> pd.DataFrame:
+        # See base class.
+
+        if not self.use_data_augmentation:
+            return df
+
+        df = df_apply_mirror_augmentation(
+            df, self._parameter_name, mirror_point=self.mirror_point
+        )
+
+        return df
+
+    @override
+    def validate_searchspace_context(self, searchspace: SearchSpace) -> None:
+        """See base class.
+
+        Args:
+            searchspace: The searchspace to validate against.
+
+        Raises:
+            TypeError: If the affected parameter is not numerical.
+        """
+        super().validate_searchspace_context(searchspace)
+
+        param = searchspace.get_parameters_by_name(self.parameter_names)[0]
+        if not param.is_numerical:
+            raise TypeError(
+                f"In the {self.__class__.__name__}, the affected parameter must be "
+                f"numerical. However, the parameter '{param.name}' is of type "
+                f"{param.__class__.__name__} and is not numerical."
+            )
+
+
+@define(frozen=True)
+class DependencySymmetry(Symmetry):
+    """Class for representing dependency symmetries.
+
+    A dependency symmetry expresses that certain parameters are dependent on another
+    parameter having a specific value. For instance, the situation "The value of
+    parameter y only matters if parameter x has the value 'on'.". In this scenario x
+    is the causing parameter and y depends on x.
```

Comment:
Comment regarding terminology, here and in other docstrings: since the name is  "DependencySymmetry" and we talk about the "dependent parameters":
* Wouldn't make sense to call the "causing" parameter the "independent parameter" instead? Would be more consistent and also solves the problem that "causing" sort of implies a causal relationship from x to y (in the sense that x physically affects y) which is not the case
* You call them "dependent" parameters, which is fine to me, but why do you then call the corresponding attribute "affected"?